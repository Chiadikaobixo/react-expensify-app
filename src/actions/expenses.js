import db from '../firebase/firebase'
import { ref, push} from 'firebase/database';

//Expense Action generator
export const addExpense = (expense) => ({
    type: 'ADD_EXPENSE',
    expense 
})

export const startAddExpense = (expenseData = {}) => {
    return (dispatch) => {
        const {
          description = '',
          note = '', 
          amount = 0, 
          createdAt = 0
        } = expenseData
        const expense = {description, note, amount, createdAt}
        return push(ref(db, 'expenses'), {
                ...expense
            }).then((ref) => {
                dispatch(addExpense({
                    id: ref.key,
                    ...expense
                }))
            })
    }
}

  
export const removeExpense = ({id} = {}) => ({
    type: 'REMOVE_EXPENSE',
    id
})
  
export const editExpense = (id, updates) => ({
    type: 'EDIT_EXPENSE',
    id,
    updates
})