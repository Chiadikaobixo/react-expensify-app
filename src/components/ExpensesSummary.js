import React from "react";
import { connect } from 'react-redux'
import getVisibleExpenses from '../selectors/expenses'
import expenseTotal from "../selectors/expenseTotal";
import { Link } from 'react-router-dom'
import numeral from 'numeral'

export const ExpensesSummary = ({ count, total }) => {
    const expenseWord = count === 1 ? 'expense' : 'expenses'
    const formatedTotal = numeral(total / 100).format('$0,0.00')
    return (
        <div className="page-header">
            <div className="content-container">
                <h1
                    className="page-header__title">viewing <span>{count}</span>
                    {expenseWord} totalling <span>{formatedTotal}</span>
                </h1>
                <div className="page-header__actions">
                    <Link className="button" to="/create">Add Expense</Link>
                </div>
            </div>
        </div>
    )
}

const mapStateToProps = (state) => {
    const visibleExpense = getVisibleExpenses(state.expenses, state.filters)
    return {
        count: visibleExpense.length,
        total: expenseTotal(visibleExpense)
    }
}
export default connect(mapStateToProps)(ExpensesSummary)