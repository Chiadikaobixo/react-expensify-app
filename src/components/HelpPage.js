import React from "react";

const HelpPage = () => (
    <div>
      this is the HelpPage!!
    </div>
)

export default HelpPage