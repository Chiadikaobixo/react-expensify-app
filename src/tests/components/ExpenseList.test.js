import React from "react";
import { shallow } from 'enzyme'
import { ExpenseList } from "../../components/ExpenseList";
import expenses from "../fixtures/expenses";

test('show render expenseList with expenses', () => {
    const wrapper = shallow(<ExpenseList expenses={expenses}/>)
    expect(wrapper).toMatchSnapshot()
})

test('show render expenseList with no expenses', () => {
    const wrapper = shallow(<ExpenseList expenses={[]}/>)
    expect(wrapper).toMatchSnapshot()
})
