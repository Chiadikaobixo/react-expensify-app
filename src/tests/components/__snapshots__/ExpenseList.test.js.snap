// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`show render expenseList with expenses 1`] = `
<div
  className="content-container"
>
  <div
    className="list-header"
  >
    <div
      className="show-for-mobile"
    >
      Expenses
    </div>
    <div
      className="show-for-desktop"
    >
      Expense
    </div>
    <div
      className="show-for-desktop"
    >
      Amount
    </div>
  </div>
  <div
    className="list-body"
  >
    <ExpenseListItem
      amount={100}
      createdAt={0}
      description="car"
      id="1"
      key="1"
      note=""
    />
    <ExpenseListItem
      amount={5000}
      createdAt={-345600000}
      description="rent"
      id="2"
      key="2"
      note=""
    />
    <ExpenseListItem
      amount={3500}
      createdAt={345600000}
      description="credit card"
      id="3"
      key="3"
      note=""
    />
  </div>
</div>
`;

exports[`show render expenseList with no expenses 1`] = `
<div
  className="content-container"
>
  <div
    className="list-header"
  >
    <div
      className="show-for-mobile"
    >
      Expenses
    </div>
    <div
      className="show-for-desktop"
    >
      Expense
    </div>
    <div
      className="show-for-desktop"
    >
      Amount
    </div>
  </div>
  <div
    className="list-body"
  >
    <div
      className="list-item list-item--message"
    >
      <span>
        No Expenses
      </span>
    </div>
  </div>
</div>
`;
